google/cloud/internal/async_read_stream_impl.h

```diff
@@ -150,10 +151,12 @@ class AsyncReadStreamImpl
      private:
       void Cancel() override {}  // LCOV_EXCL_LINE
       bool Notify(bool ok) override {
+        OptionsSpan span(options_);
```

Comment:
I wonder if we should set these spans in the completion queue instead.  The `Start()` function is a good place to capture the current options:

https://github.com/googleapis/google-cloud-cpp/blob/b92933ba81c2e497e7daf239ce9872080da2828e/google/cloud/internal/default_completion_queue_impl.cc#L238-L241

and the notifications would be a good place to create the span:

https://github.com/googleapis/google-cloud-cpp/blob/b92933ba81c2e497e7daf239ce9872080da2828e/google/cloud/internal/default_completion_queue_impl.cc#L175-L179

Reply:
I considered that, but thought that changing the `tag` map (which is really just a collection of `AsyncGrpcOperation`s) to a map of `AsyncGrpcOperation`/`Options` pairs was both a little more complicated, and belied the fact that the `Options` logically belonged to the operation.

It is certainly the case that the behavior is now distributed to the `AsyncGrpcOperation` sub-classes, but I've been thinking of that as a feature.  For example, the existence of the `internal::CompletionQueueImpl` interface implies that this behavior should not belong in a particular instantiation (the `DefaultCompletionQueueImpl`) of that interface.

Perhaps experience with this will lead to a better understanding, but for now I'd vote for leaving it as is.

Reply:
> For example, the existence of the `internal::CompletionQueueImpl` interface implies that this behavior should not belong in a particular instantiation (the `DefaultCompletionQueueImpl`) of that interface.

FWIW, the only reason for the interface is mocking :shrug:

> Perhaps experience with this will lead to a better understanding, but for now I'd vote for leaving it as is.

SGTM.